go/pkg/tree/tree.go

```diff
@@ -99,15 +109,15 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 		return nil
 	}
 	for _, f := range files {
-		if e := loadFiles(execRoot, excl, filepath.Join(path, f.Name()), fs, chunkSize, cache); e != nil {
+		if e := loadFiles(execRoot, excl, filepath.Join(path, f.Name()), fs, chunkSize, cache, config); e != nil {
 			return e
 		}
 	}
 	return nil
 }
 
 // ComputeMerkleTree packages an InputSpec into uploadable inputs, returned as Chunkers.
-func ComputeMerkleTree(execRoot string, is *command.InputSpec, chunkSize int, cache filemetadata.Cache) (root digest.Digest, inputs []*chunker.Chunker, stats *Stats, err error) {
+func ComputeMerkleTree(execRoot string, is *command.InputSpec, chunkSize int, cache filemetadata.Cache, config ConstructTreeConfig) (root digest.Digest, inputs []*chunker.Chunker, stats *Stats, err error) {
```

Comment:
Is it possible to pass each config as optional args by making this variadic function?

Reply:
I believe that's going to restrict the config params to be of the same type?

If your concern is around the temporary break of `cas`, we can add another function, e.g. `ComputeMerkleTreeWithConfig`?

Reply:
I imagined to use functional option pattern, but `ComputeMerkleTreeWithConfig` is fine too.

Reply:
> I imagined to use functional option pattern

Ah, I guess you mean something like

```go
type ConstructOption = func(ConstrucConfig*)

func ComputeMerkleTree(execRoot string, is *command.InputSpec, chunkSize int, cache filemetadata.Cache,  opts ... ConstructOption) {
  config := &ConstrucConfig{}
  for o := range opts {
    o(config)
  }
  ...
}
```

That seems cleaner.

Reply:
yes.

Reply:
Done